src/com/amazon/corretto/crypto/provider/AesCbcSpi.java

```diff
@@ -372,6 +377,8 @@ private int update(
       final byte[] outputArray,
       final int outputOffset) {
 
+    inputIsEmpty = inputIsEmpty && (inputLen == 0);
```

Comment:
nit: I think it'd clearer if the logic looked like this. No need to reason about the state of inputIsEmpty between invocations.

```
if (inputLen > 0) {
  inputIsEmpty = false;
}
```

Reply:
If I don't get any other comments, I'll have it done in the PR that is going to refactor the scratch pad length log, which I promised.

Reply:
is the idea here to track `inputIsEmpty` state across **all** update/doFinal calls, and if _any_ of them have non-empty inputs, ensure we go through full processing logic in doFinal?

Reply:
if so, do we also need to reset `inputIsEmpty` in [engineInit](https://github.com/corretto/amazon-corretto-crypto-provider/blob/main/src/com/amazon/corretto/crypto/provider/AesCbcSpi.java#L232)?